src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,35 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
+    private static final DateTimeFormatter DATE_TIME_FORMATTER
+            = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+    private static final String SEPARATOR = " ";
+    private static final long AND_ONE_MORE_DAY = 1L;
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        StringBuilder stringBuilder = new StringBuilder("Report for period ");
+        stringBuilder.append(dateFrom).append(" - ").append(dateTo).append(System.lineSeparator());
+        for (String name : names) {
+            int salary = 0;
+            for (String record : data) {
+                var separatedData = record.split(SEPARATOR);
+                var employeeData = LocalDate.parse(separatedData[0], DATE_TIME_FORMATTER);
+                var employeeName = separatedData[1];
+                var workingOurs = Integer.parseInt(separatedData[2]);
+                var incomePerOur = Integer.parseInt(separatedData[3]);
```

Comment:
![image](https://user-images.githubusercontent.com/12864128/201327887-38f26546-ed64-4fa9-afa3-f1922247f1ed.png)
Move index to the [constant](https://mate-academy.github.io/style-guides/java/java.html#s5.2.4-constant-namesl)

Reply:
My bad. Refactored